src/core/dataProviders/dataProvider.js

```diff
@@ -249,12 +249,19 @@ const convertHTTPResponse = (response, type, resource, params) => {
 
         case UPDATE:
             return params;
+
         case CREATE:
             const dataFromRequest = get(params, 'data', null);
             const compositionUid = get(response, 'compositionUid', null);
-            const compositionUidArray = compositionUid.split('::');
-            const sourceID = compositionUidArray[0];
+            let sourceID = '';
+            if (compositionUid) {
+                const compositionUidArray = compositionUid.split('::');
+                sourceID = compositionUidArray[0];
+            }
             dataFromRequest.id = get(response, 'host', null) + '-' + sourceID;
+            if (!get(params, 'source', null)) {
+                dataFromRequest.source = 'ethercis';
```

Comment:
it's not a good practice to mutate parameters - it can have unexpected side effects.You should create a new object instead:

```js
return { data: { ...dataFromRequest, source: 'ethercis' } }:
```